src/chainstate/burn/db/sortdb.rs

```diff
@@ -1424,14 +1430,47 @@ impl<'a> SortitionHandleConn<'a> {
             // no winner
             return Ok(vec![]);
         }
+        let qry = "SELECT * FROM block_commits WHERE sortition_id = ?1 AND txid = ?2";
+        let args: [&dyn ToSql; 2] = [&snapshot.sortition_id, &snapshot.winning_block_txid];
+        let winning_commit: LeaderBlockCommitOp = query_row(self, qry, &args)?
+            .expect("BUG: sortition exists, but winner cannot be found");
 
         let winning_block_hash160 =
             Hash160::from_sha256(snapshot.winning_stacks_block_hash.as_bytes());
 
-        let qry = "SELECT * FROM user_burn_support WHERE sortition_id = ?1 AND block_header_hash_160 = ?2 ORDER BY vtxindex ASC";
-        let args: [&dyn ToSql; 2] = [&snapshot.sortition_id, &winning_block_hash160];
+        let qry = "SELECT * FROM user_burn_support
+                   WHERE sortition_id = ?1 AND block_header_hash_160 = ?2 AND key_vtxindex = ?3 AND key_block_ptr = ?4
+                   ORDER BY vtxindex ASC";
+        let args: [&dyn ToSql; 4] = [
+            &snapshot.sortition_id,
+            &winning_block_hash160,
+            &winning_commit.key_vtxindex,
+            &winning_commit.key_block_ptr,
+        ];
+
+        let mut winning_user_burns: Vec<UserBurnSupportOp> = query_rows(self, qry, &args)?;
+
+        // were there multiple miners with the same VRF key and block header hash? (i.e., are these user burns shared?)
+        let qry = "SELECT COUNT(*) FROM block_commits
+                   WHERE sortition_id = ?1 AND block_header_hash = ?2 AND key_vtxindex = ?3 AND key_block_ptr = ?4";
+        let args: [&dyn ToSql; 4] = [
+            &snapshot.sortition_id,
+            &snapshot.winning_stacks_block_hash,
+            &winning_commit.key_vtxindex,
+            &winning_commit.key_block_ptr,
+        ];
+        let shared_miners = query_count(self, qry, &args)? as u64;
 
-        query_rows(self, qry, &args)
+        assert!(
+            shared_miners >= 1,
+            "BUG: Should be at least 1 matching miner for the winning block commit"
+        );
+
+        for winning_user_burn in winning_user_burns.iter_mut() {
```

Comment:
I would prefer it if this adjustment happened within the sortition distribution logic, instead of in the sortition DB.

Reply:
This method is only ever invoked by the anchor block processing -- I could update this to return the original block commit ops and the number of block commits that were identified by the same user burns. Then the caller (`preprocess_anchor_block`) would be responsible for applying the adjustment before storing the staging user burns. Otherwise, I think this would entail a lot of refactoring of how the miner rewards are calculated.

Reply:
Ah, I see.